workers/task-driver/src/tasks/create_new_wallet.rs

```diff
@@ -301,4 +310,16 @@ impl NewWalletTask {
             },
         )
     }
+
+    /// Emit a wallet creation event to the event manager
+    fn emit_event(&self) -> Result<(), NewWalletTaskError> {
+        let event = RelayerEvent::WalletCreation(WalletCreationEvent {
```

Comment:
we can probably provide constructors that fill in obvious fields like event id and event timestamp and only require those fields not in the scope of the constructor